llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -515,6 +515,7 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
   getActionDefinitionsBuilder(G_ICMP)
       .legalFor({{s32, s32},
                  {s32, s64},
+                 {s64, s64},
```

Comment:
I think this and the select below is marking a s64 condition code as legal, which Im not sure is what we would want. We have the clampScalar(0, s32, s32) below.

Reply:
Well, s64 case doesn't work without this. Why don't want this to be legal?

Reply:
{s64, s64} is legal for G_SELECT but further the index is clamped to s32. I don't understand why this is done. 
Inspiring from this I made {s64, s64} legal for G_ICMP.